### PR TITLE
fix for residual acquisitions

### PR DESCRIPTION
**Pull request by visagim**
- [ ] check that fix solves the problem

### CODE REVIEW

**Comment by visagim** [comment]
This leaves some issues hanging, added a task to the backlog
https://www.notion.so/qilimanjaro/Optimize-cache-in-qblox-modules-7e916bde331242e2af2ccf5cd7ad91c1

File: src/qililab/instruments/qblox/qblox_qrm.py [hunk 1]
```diff
@@ -147,6 +147,17 @@ def compile(
...
+
+        # empty cache and sequence for specific qubits if they are not in the schedule
+        unused_seq = []
+        for cached_qubit, seq_id in (
+            (timeline.qubit, key) for key, element in self._cache.items() for timeline in element.timeline
+        ):
+            if cached_qubit not in (schedule.qubit for schedule in qubit_schedules):
+                unused_seq.append(seq_id)
+                _ = self.sequences.pop(seq_id)
+        self._cache = {key: items for key, items in self._cache.items() if key not in unused_seq}
```
Comment (AlbertMitjans) [inline src/qililab/instruments/qblox/qblox_qrm.py:159]: Couldn't we do this?
```suggestion
        for cached_qubit, seq_id in (
            (timeline.qubit, key) for key, element in self._cache.items() for timeline in element.timeline
        ):
            if cached_qubit not in (schedule.qubit for schedule in qubit_schedules):
                unused_seq.append(seq_id)
                _ = self.sequences.pop(seq_id)
                _ = self._cache.pop(seq_id)
```

**Review by AlbertMitjans — APPROVED** [review]
Good job! Added one minor comment

File: src/qililab/instruments/qblox/qblox_qrm.py (hunk 1 above)
Comment (visagim) [reply src/qililab/instruments/qblox/qblox_qrm.py]: yup

**Pull request merged**